routing_model/src/flows_dst.rs

```diff
@@ -278,7 +280,15 @@ impl CheckAndProcessElderChange {
 
     fn try_consensus(&self, vote: &ParsecVote) -> Option<Self> {
         if ParsecVote::CheckElder == *vote {
```

Comment:
I think you'd be better off extracting the ProcessElderChange sub routine first.
You can see an example of sub routine in the source or destination flows, but feel free to implement the sub-routine differently if it make sense (these other sub routine will be removed to match the new graph, but the elder/merge/split will remain).

Reply:
This would allow you to refactor and all the test should still pass.